### ISSUE TITLE
reinstate CI

### ISSUE THREAD

**Issue by agarwal** [biocaml/biocaml#168]
We long ago removed Travis CI support (see 8eec2354d17b02d01475c1686b0eec94aca0c9b6). Would be worth reinstating.

**Comment by kanishka-azimi** [comment]
I am personally interested in self hosted buildbot instance since buildbot has the strongest free license. Generally, I am aware of open source projects using some free tier from circle, azure, github actions, or using gitlab and it's integrated ci. I would vote for gitlab if people don't like buildbot, and up for whatever the group prefers overall.

**Comment by agarwal** [comment]
I'm not too motivated by this issue. In my experience, maintaining CI for a project like this has been far more work than worth it. What would be good is for `dune runtest` to do something useful, i.e. have useful tests that run fast. Developers can simply run that command when they want.

Writing tests can also be a waste of time unless we have some confidence about what to test and know that won't change too soon. I think that is achievable in this project since the signatures are pretty stable.

**Comment by kanishka-azimi** [comment]
Yes, I would delay CI for a long time, until commit activity and number of developers was much higher. Locally running any tests will provide the enough feedback at this activity level.

Keeping clear instructions in the README for how to run tests is all that I would have for now.
